src/transformers/trainer.py

```diff
@@ -336,13 +336,14 @@ def _setup_wandb(self):
             WANDB_DISABLED:
                 (Optional): boolean - defaults to false, set to "true" to disable wandb entirely
         """
-        logger.info('Automatic Weights & Biases logging enabled, to disable set os.environ["WANDB_DISABLED"] = "true"')
-        wandb.init(project=os.getenv("WANDB_PROJECT", "huggingface"), config=vars(self.args))
-        # keep track of model topology and gradients
-        if os.getenv("WANDB_WATCH") != "false":
-            wandb.watch(
-                self.model, log=os.getenv("WANDB_WATCH", "gradients"), log_freq=max(100, self.args.logging_steps)
-            )
+        if (is_tpu_available() and self.is_world_master()) or not is_tpu_available():
```

Comment:
@julien-c haven't tried with DP/DDP so I don't know if it should just be `self.is_world_master()`.

Have you already tried wandb on such setups before?

Reply:
this is also a problem outside of tpu:
https://github.com/wandb/client/issues/452#issuecomment-557779621

Reply:
@sshleifer see https://github.com/huggingface/transformers/pull/4851#issuecomment-641587040

Reply:
good link, thanks for linking